cli/admin/src/test/java/io/pravega/cli/admin/dataRecovery/DataRecoveryTest.java

```diff
@@ -0,0 +1,490 @@
+            //create stream
+            Boolean createStreamStatus = streamManager.createStream(scopeName, streamName, config);
+            log.info("Create stream status {}", createStreamStatus);
+        }
+    }
+
+    /**
+     * Tests DurableLog recovery command.
+     * @throws Exception    In case of any exception thrown while execution.
+     */
+    @Test
+    public void testDataRecoveryCommand() throws Exception {
+        int instanceId = 0;
+        int bookieCount = 3;
+        int containerCount = 1;
+        @Cleanup
+        PravegaRunner pravegaRunner = new PravegaRunner(instanceId++, bookieCount, containerCount, this.storageFactory);
+        String streamName = "testDataRecoveryCommand";
+
+        createScopeStream(pravegaRunner.controllerRunner.controller, SCOPE, streamName);
+        try (val clientRunner = new ClientRunner(pravegaRunner.controllerRunner)) {
+            // Write events to the streams.
+            writeEvents(streamName, clientRunner.clientFactory);
+        }
+        pravegaRunner.controllerRunner.close(); // Shut down the controller
+
+        // Flush all Tier 1 to LTS
+        ServiceBuilder.ComponentSetup componentSetup = new ServiceBuilder.ComponentSetup(pravegaRunner.segmentStoreRunner.serviceBuilder);
+        for (int containerId = 0; containerId < containerCount; containerId++) {
+            componentSetup.getContainerRegistry().getContainer(containerId).flushToStorage(TIMEOUT).join();
+        }
+
+        pravegaRunner.segmentStoreRunner.close(); // Shutdown SegmentStore
+        pravegaRunner.bookKeeperRunner.close(); // Shutdown BookKeeper & ZooKeeper
+
+        // start a new BookKeeper and ZooKeeper.
+        pravegaRunner.bookKeeperRunner = new BookKeeperRunner(instanceId++, bookieCount);
+
+        // set pravega properties for the test
+        STATE.set(new AdminCommandState());
+        Properties pravegaProperties = new Properties();
+        pravegaProperties.setProperty("pravegaservice.container.count", "1");
+        pravegaProperties.setProperty("pravegaservice.storage.impl.name", "FILESYSTEM");
+        pravegaProperties.setProperty("pravegaservice.storage.layout", "ROLLING_STORAGE");
+        pravegaProperties.setProperty("filesystem.root", this.baseDir.getAbsolutePath());
+        pravegaProperties.setProperty("pravegaservice.zk.connect.uri", "localhost:" + pravegaRunner.bookKeeperRunner.bkPort);
+        pravegaProperties.setProperty("bookkeeper.ledger.path", pravegaRunner.bookKeeperRunner.ledgerPath);
+        pravegaProperties.setProperty("bookkeeper.zk.metadata.path", pravegaRunner.bookKeeperRunner.logMetaNamespace);
+        pravegaProperties.setProperty("pravegaservice.clusterName", pravegaRunner.bookKeeperRunner.baseNamespace);
+        STATE.get().getConfigBuilder().include(pravegaProperties);
+
+        // Command under test
+        TestUtils.executeCommand("storage DurableLog-recovery", STATE.get());
```

Comment:
wait, why is there a capital `D` in this command? I thought all commands were lowercase.

Reply:
made it small `d`.